scripts/core/log_safety.py

```
@@ -0,0 +1,140 @@
+
+    Hostile objects may raise from ``__str__`` or return a non-str. We
+    do not fall back to ``repr()`` because ``repr()`` can also raise on
+    hostile objects; a fixed sentinel is safer.
+    """
+    if value is None:
+        return _NONE_MARKER
+    if isinstance(value, str):
+        return value
+    try:
+        result = str(value)
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)